eth/transactor.go

```diff
@@ -227,16 +205,116 @@ func (t *Transactor) transact(
 	}
 }
 
-func determineGasPrice(txopts txOptions, client *ethclient.Client) (*big.Int, error) {
+// determineGas sets the gas price and gas limit on the signer
+func (t *Transactor) determineGas(method TxMethod, signer *bind.TransactOpts, txopts txOptions, client *ethclient.Client) error {
+	// 1. Determine gas price
+	// Only accept legacy flags or EIP-1559 flags, not both
+	hasLegacyFlags := txopts.forceGasGwei > 0 || txopts.minGasGwei > 0 || txopts.maxGasGwei > 0 || txopts.addGasGwei > 0
+	has1559Flags := txopts.maxFeePerGasGwei > 0 || txopts.maxPriorityFeePerGasGwei > 0
+	if hasLegacyFlags && has1559Flags {
+		return ErrConflictingGasFlags
+	}
+	// Check if chain supports EIP-1559
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	head, err := client.HeaderByNumber(ctx, nil)
```

Comment:
should we consider move eip1559 into Transactor and set true/false when new transactor?
this saves queries in each tx
the down side is in theory a chain could become eip1559 after certain block number, in our case we'll just keep legacy until next restart, so seems fine.

Reply:
Ok with this. Note that go-ethereum will do this query regardless.

Reply:
never mind, I see you need head.BaseFee later